src/win/thread.c

```diff
@@ -191,7 +191,13 @@ int uv_thread_create(uv_thread_t *tid, void (*entry)(void *arg), void *arg) {
 
 
 uv_thread_t uv_thread_self(void) {
-  return (uv_thread_t) uv_key_get(&uv__current_thread_key);
+  void* self;
+  uv_once(&uv__current_thread_init_guard, uv__init_current_thread_key);
+  self = uv_key_get(&uv__current_thread_key);
+  if (self == NULL)
+      /* This thread wasn't created with uv_thread_create. */
+      self = GetCurrentThread();
```

Comment:
Hum. The problem is that it would be NULL for any thread not created by libuv, not only the main thread. Right now, by accident or not, on Unix we return a valid thread id even if we didn't create the thread.

> Note that GetCurrentThread() always returns a pseudo-thread handle which is only valid in the context of the current thread, and if I remember correctly, the same for all threads.

Since we don't store it, wouldn't this suffice?